### ISSUE TITLE
ww_multi_scale returns an error for categorical rasters

### ISSUE THREAD

**Issue by Nowosad** [ropensci/waywiser#60]
Hi @mikemahoney218 -- I am trying to incorporate **waywiser** for examples in a paper about spatial patterns comparison. The `ww_multi_scale()` function works fine for continuous rasters (however, it gives some CRSs warnings), but errors for categorical ones. Is this expected?

``` r
library(terra)
library(waywiser)
library(landscapemetrics)
cell_sizes = seq(10, 100, 10) * 1000
l1 = rast(landscape)
l2 = l1
clc_multi_scale = ww_multi_scale(truth = l1, estimate = l2,
                                 metrics = list(yardstick::precision), 
                                 cellsize = cell_sizes)
#> Error in `metric_set()`:
#> ! Failed to compute `<clss_mtr>()`.
#> Caused by error:
#> ! Must select at least one item.
clc_multi_scale
#> Error in eval(expr, envir, enclos): object 'clc_multi_scale' not found
```

**Comment by mikemahoney218** [comment]
Thanks for the report, I can reproduce on my end.

I'm digging into this now, and am going to write notes here as I go. It seems like for some reason we're not passing `estimate` to the yardstick functions correctly in this case:
```
Browse[3]> enquo(truth)
<quosure>
expr: ^.truth
env:  0x1634bae58
Browse[3]> enquo(estimate)
<quosure>
expr: ^
env:  empty
```

**Comment by mikemahoney218** [comment]
on a second look, while I can't confirm that's _not_ a problem, I think the bigger issue is that these columns aren't factors when working with classification metrics: https://github.com/tidymodels/yardstick/issues/504

**Comment by mikemahoney218** [comment]
Thanks for the report! I think this is mostly fixed in #61. The only other change necessary (and you can see this in the tests added in #61) is that those `cellsizes` wind up creating a grid with a single cell that includes the entire raster; as `precision()` requires factors have more than one level, this winds up erroring. Using a smaller grid will work once #61 is merged though.
